crates/pallet-executor/src/lib.rs

```diff
@@ -455,6 +551,24 @@ impl<T: Config> Pallet<T> {
         Ok(())
     }
 
+    fn validate_fraud_proof(fraud_proof: &FraudProof) -> Result<(), Error<T>> {
+        let to_prove: T::BlockNumber = (fraud_proof.parent_number + 1u32).into();
+        ensure!(
+            to_prove >= OldestReceiptNumber::<T>::get()
+                && to_prove <= ExecutionChainBestNumber::<T>::get(),
+            Error::<T>::ExecutionReceiptPrunedOrFuture
```

Comment:
Out of curiosity, why separate error variants for ERs and one for FP?

Reply:
Because I didn't find a satisfying naming for `ExecutionReceiptTooFarInFutureForFraudProof` 🙈  Any suggestion?

Reply:
I meant why not `ExecutionReceiptPruned` and `ExecutionReceiptInFuture` separately :shrug:

Reply:
`ExecutionReceiptTooFarInFuture` has already been used for `submit_execution_receipt`, using `ExecutionReceiptInFuture` for FP is not enough to distinguish these two and I'm not sure it's a good idea to use one kind for two cases even they share some similarity. https://github.com/subspace/subspace/blob/64afe53c02a546872a4ae20e10d2edbf76fe8e62/crates/pallet-executor/src/lib.rs#L540

Reply:
Thanks for the clue, the nested error looks much better!